authorizer-app/src/app/shared/containers/authorization-complete-page/authorization-complete-page.component.ts

```diff
@@ -27,7 +27,22 @@ export class AuthorizationCompletePageComponent implements OnInit {
 
   ngOnInit(): void {
     this.isLoading = true;
-    const {state, oauth_token, oauth_verifier, oauth_token_secret, code} = this.activatedRoute.snapshot.queryParams;
+    const queryParams = this.activatedRoute.snapshot.queryParams;
+    const storedParams = this.service.getUserAuthParams();
+    const state = this.getOrDefault(queryParams.state, storedParams.state);
+    const oauth_token = this.getOrDefault(
+      queryParams.oauth_token,
+      storedParams.oauth_token
+    );
+    const oauth_verifier = this.getOrDefault(
+      queryParams.oauth_verifier,
+      storedParams.oauth_verifier
+    );
+    const oauth_token_secret = this.getOrDefault(
+      queryParams.oauth_token_secret,
+      storedParams.oauth_token_secret
+    );
```

Comment:
Should we delete these items from storage after the authorization is complete to avoid it interfering with future registrations?